akka-cluster/src/main/scala/akka/cluster/MembershipState.scala

```diff
@@ -119,4 +124,137 @@ import akka.annotation.InternalApi
       .map(_.uniqueAddress)
   }
 
+  def validNodeForGossip(node: UniqueAddress): Boolean =
+    node != selfUniqueAddress && isReachableExcludingDownedObservers(node)
+
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] class GossipTargetSelector(val crossDcConnections: Int, val reduceGossipDifferentViewProbability: Double) {
+
+  final def gossipTarget(state: MembershipState): Option[UniqueAddress] = {
+    selectRandomNode(gossipTargets(state))
+  }
+
+  final def gossipTargets(state: MembershipState): Vector[UniqueAddress] =
+    if (state.latestGossip.isMultiDc) multiDcGossipTargets(state)
+    else localDcGossipTargets(state)
+
+  /**
+   * Chooses a set of possible gossip targets that is in the same dc. If the cluster is not multi dc this will
+   * means it is a choice among all nodes of the cluster.
+   */
+  protected def localDcGossipTargets(state: MembershipState): Vector[UniqueAddress] = {
+    val latestGossip = state.latestGossip
+    val firstSelection: Vector[UniqueAddress] =
+      if (preferNodesWithDifferentView(state)) {
+        // If it's time to try to gossip to some nodes with a different view
+        // gossip to a random alive same dc member with preference to a member with older gossip version
+        latestGossip.members.collect {
+          case m if m.dataCenter == state.selfDc && !latestGossip.seenByNode(m.uniqueAddress) && state.validNodeForGossip(m.uniqueAddress) ⇒
+            m.uniqueAddress
+        }(breakOut)
+      } else Vector.empty
+
+    // Fall back to localGossip
+    if (firstSelection.isEmpty) {
+      latestGossip.members.toVector.collect {
+        case m if m.dataCenter == state.selfDc && state.validNodeForGossip(m.uniqueAddress) ⇒ m.uniqueAddress
+      }
+    } else firstSelection
+
+  }
+
+  /**
+   * Choose cross-dc nodes if this one of the N oldest nodes, and if not fall back to gosip locally in the dc
+   */
+  protected def multiDcGossipTargets(state: MembershipState): Vector[UniqueAddress] = {
+    val latestGossip = state.latestGossip
+    // 20% of the times across dcs and 80% local (doing it for all nodes to avoid having to collect the oldest nodes per
+    // dc all the time, when #23290 is merged we can maybe cache oldest members per dc in that)
```

Comment:
I realised something here: I was taking up to the max nr of interconnections when creating the map. But I think it should really be all nodes per DC sorted by age in the map since filtering is done on whether the nodes are suitable for gossip, and the number limit is among the first N "gossipable" nodes, WDYT?